src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OtlpMtlsCertificateManager.cs

```diff
@@ -0,0 +1,352 @@
+                            StringComparison.OrdinalIgnoreCase))
+                    {
+                        OpenTelemetryProtocolExporterEventSource.Log.MtlsServerCertificateValidated(
+                            serverCert.Subject);
+                        return true;
+                    }
+                }
+            }
+
+            OpenTelemetryProtocolExporterEventSource.Log.MtlsServerCertificateValidationFailed(
+                serverCert.Subject,
+                sslPolicyErrors.ToString());
+
+            return false;
+        }
+        catch (Exception ex)
+        {
+            OpenTelemetryProtocolExporterEventSource.Log.MtlsServerCertificateValidationFailed(
+                serverCert.Subject,
+                ex.Message);
+
+            return false;
+        }
+    }
+
+    private static void ValidateFileExists(string filePath, string fileType)
+    {
+        if (string.IsNullOrEmpty(filePath))
+        {
+            throw new ArgumentException(
+                $"{fileType} path cannot be null or empty.",
+                nameof(filePath));
+        }
+
+        if (!File.Exists(filePath))
+        {
+            OpenTelemetryProtocolExporterEventSource.Log.MtlsCertificateFileNotFound(
+                fileType,
+                filePath);
+            throw new FileNotFoundException($"{fileType} file not found at path: {filePath}", filePath);
+        }
+    }
+
+    /// <summary>
+    /// Gets the X509RevocationMode from configuration or returns the default value.
+    /// </summary>
+    /// <param name="configuration">Configuration to read from.</param>
+    /// <returns>The configured revocation mode or default (Online).</returns>
+    private static X509RevocationMode GetRevocationModeFromConfiguration(IConfiguration? configuration)
+    {
+        if (configuration == null)
+        {
+            return X509RevocationMode.Online;
+        }
+
+        if (configuration.TryGetStringValue(OtlpSpecConfigDefinitions.CertificateRevocationModeEnvVarName, out var modeString))
+        {
+            if (Enum.TryParse<X509RevocationMode>(modeString, true, out var mode))
+            {
+                return mode;
+            }
+
+            ((IConfigurationExtensionsLogger)OpenTelemetryProtocolExporterEventSource.Log).LogInvalidConfigurationValue(
+                OtlpSpecConfigDefinitions.CertificateRevocationModeEnvVarName,
+                modeString);
+        }
+
+        return X509RevocationMode.Online;
+    }
+
+    /// <summary>
+    /// Gets the X509RevocationFlag from configuration or returns the default value.
+    /// </summary>
+    /// <param name="configuration">Configuration to read from.</param>
+    /// <returns>The configured revocation flag or default (ExcludeRoot).</returns>
+    private static X509RevocationFlag GetRevocationFlagFromConfiguration(IConfiguration? configuration)
+    {
+        if (configuration == null)
+        {
+            return X509RevocationFlag.ExcludeRoot;
```

Comment:
As this is the non-default value of this enum, can we add a comment explaining why this is the default behaviour?

Reply:
Good point, resolved!